core/src/main/scala/org/broadinstitute/dsde/rawls/workspace/WorkspaceService.scala

```diff
@@ -2735,6 +2738,10 @@ class WorkspaceService(protected val ctx: RawlsRequestContext,
         petKey,
         expectedGoogleBucketPermissions
       )
+      _ <- ApplicativeThrow[Future].raiseWhen(useDefaultPet && expectedGoogleProjectPermissions.nonEmpty) {
+        new RawlsException("user has workspace read-only access yet has expected google project permissions")
+      }
+
```

Comment:
I am torn about throwing this exception. On the one hand, it is an unexpected case. On the other hand, this will probably lead to unanticipated failure if we ever choose to add google project permissions to read-only users.

The alternative is that we just don't do the google project access check for read-only users. But that also means that if we ever do add google project permissions we will silently not be checking them.

Opinions welcome.

Reply:
I prefer to throw the exception instead of silently not checking them; it seems like that's a sure programmer error we'd  want to catch in testing should the day arrive when we do add the google project perms for readonly users.